src/app/ReadClient.cpp

```diff
@@ -1229,7 +1229,7 @@ CHIP_ERROR ReadClient::EstablishSessionToPeer()
     ChipLogProgress(DataManagement, "Trying to establish a CASE session for subscription");
     auto * caseSessionManager = InteractionModelEngine::GetInstance()->GetCASESessionManager();
     VerifyOrReturnError(caseSessionManager != nullptr, CHIP_ERROR_INCORRECT_STATE);
-    caseSessionManager->FindOrEstablishSession(mPeer, &mOnConnectedCallback, &mOnConnectionFailureCallback);
+    caseSessionManager->FindOrEstablishSession(mPeer, &mOnConnectedCallback, &mOnConnectionFailureCallback, nullptr);
```

Comment:
See, none of these changes outside CASESessionManager and OperatonalSessionSetup should be needed if we were not breaking the API.